ipaplatform/osinfo.py

```diff
@@ -208,6 +209,16 @@ def platform(self):
         raise ImportError('No ipaplatform available for "{}"'.format(
                           ', '.join(self.platform_ids)))
 
+    @property
+    def container(self):
+        if self._container is not None:
+            return self._container
+        try:
+            from ipaplatform.tasks import tasks
+            self._container = tasks.detect_container()
+            return self._container
+        except ImportError:
+            raise ImportError('Platform does not support detecting containers')
```

Comment:
* import of tasks should never fail (or we have a more serious issue)
* base task's ``detect_container`` raises ``NotImplementedError``

```
        from ipaplatform.tasks import tasks
        try:
            self._container = tasks.detect_container()
        except NotImplementedError:
            raise NotImplementedError('Platform does not support detecting containers') from None
```